dask/dataframe/utils.py

```diff
@@ -557,6 +576,27 @@ def assert_eq(
         a = a.to_pandas()
     if hasattr(b, "to_pandas"):
         b = b.to_pandas()
+
+    # Temporary changes to look for pyarrow dtype failures
+    import dask
+
+    if dask.config.get("dataframe.convert_string"):
+        from dask.dataframe._pyarrow import to_pyarrow_string
+
+        if isinstance(a, (pd.DataFrame, pd.Series, pd.Index)):
+            a = to_pyarrow_string(a)
+
+        if isinstance(b, (pd.DataFrame, pd.Series, pd.Index)):
+            b = to_pyarrow_string(b)
+
+        if (
+            isinstance(a, pd.Series)
+            and is_bool_dtype(a.dtype)
+            and isinstance(b, pd.Series)
+            and is_bool_dtype(b.dtype)
+        ):
+            check_dtype = False
```

Comment:
Hrm, I'm curious what prompted this change

Reply:
It was this test failure:

```
FAILED dask/dataframe/tests/test_accessors.py::test_str_accessor - AssertionError: Attributes of Series are different

Attribute "dtype" are different
[left]:  boolean
[right]: bool
```

I made the change in `assert_eq` projecting more tests might fai for the same reason. After all the changes are in, I re-checked and it's just this test. I reverted the change in `assert_eq` and made changes in the test instead.